src/lib/api-client.js

```diff
@@ -115,3 +115,11 @@ export function searchStudent(id){
         Dispatcher.handleAction('NO_STUDENT_FOUND');
     });
 }
+
+export function getAllModels(){
+    get('model/all', {}
+    ).then(data => {
+        Dispatcher.handleAction('MODELS_RECEIVED', data);
+        hashHistory.push('/models');
```

Comment:
This is fine as it is, I think. Stores only contain state and consume actions to update state. Routing to another page is somewhat outside of their scope.